spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc

```diff
@@ -1159,4 +1160,72 @@ Spring Cloud Data Flow will have a suffix appended to it consisting of `-scdf-<t
+
+There are three main flows to consider in the above diagram.  Launching the first time/launching with no changes is one, the other is launching when there are changes.  We'll look at the flow with no changes first.
+
+==== Launch a Task With No Changes
+1. A launch request comes into to Data Flow.  Data Flow determines that an upgrade is not required since nothing has changed (no properties, deployment properites, or versions have changed since the last execution).
+
+[start=5]
+5. On platforms that cache a deployed artifact (CloudFoundry at the writing of this documentation), Data Flow checks if the application was previously deployed.
+6. If the application needs to be deployed, Data Flow will do the deployment of the task application.
+7. Data Flow launches the application.
+
+That flow is the default behavior and will occur every time a request comes in if nothing has changed.  It is important to note that this is the same flow that Data Flow has always executed for launching of tasks.
+
+==== Launch a Task With Changes That Is Not Currently Running
+The second flow to consider when launching a task is if there was a change in any of the task application version, application properties, or deployment properties.  In this case, the following flow is executed
+
+1. A launch request comes into to Data Flow.  Data Flow determines that an upgrade is required since there was a change in either task application version, application properties, or deployment properties.
+2. Data Flow checks to see if another instance of the task definition is currently running.
+
+[start=4]
+4. If there is not another instance of the task definition currently running, the old deployment is deleted.
+5. On platforms that cache a deployed artifact (CloudFoundry at the writing of this documentation), Data Flow checks if the application was previously deployed (this check will always be no in this flow).
+6. Data Flow will do the deployment of the task application with the updated values (new application version, new merged properties, new merged deployment properties).
+7. Data Flow launches the application.
+
+This flow is what fundementally enables continuous deployment for Spring Cloud Data Flow.
+
+==== Launch a Task With Changes While Another Instance Is Running
+The last main flow is when a launch request comes to Spring Cloud Data Flow to do an upgrade but the task definition is currently running.  In this case, the launch is blocked due to the requirement to delete the current application.  On some platforms (CloudFoundry at the writing of this document), deleting the application will cause all currently running applications to be shut down.  This feature prevents that from happening.
+
+1. A launch request comes into to Data Flow.  Data Flow determines that an upgrade is required since there was a change in either task application version, application properties, or deployment properties.
```

Comment:
This process description should be preceded by a leader sentence -- something like "The following process describes what happens when a task changes while another instance is running:" (note the colon at the end).

Also, add a comma between "required" and "since".